index-protocol/src/Ergvein/Index/Protocol/Types.hs

```diff
@@ -98,6 +100,36 @@ word32ToCurrencyCode = \case
   13 -> Just TDASH
   _  -> Nothing
 
+codeToCurrency :: CurrencyCode -> Maybe E.Currency
+codeToCurrency = \case
+  BTC    -> Just E.BTC
+  TBTC   -> Just E.BTC
+  ERGO   -> Just E.ERGO
+  TERGO  -> Just E.ERGO
+  _ -> Nothing
+
+currencyToCode :: Bool -> E.Currency -> CurrencyCode
+currencyToCode test = \case
+  E.BTC -> if test then TBTC else BTC
+  E.ERGO -> if test then TERGO else ERGO
+
+currencyCodeTestnet :: CurrencyCode -> Bool
+currencyCodeTestnet = \case
+  BTC    -> False
+  TBTC   -> True
+  ERGO   -> False
+  TERGO  -> True
+  USDTO  -> False
+  TUSDTO -> True
+  LTC    -> False
+  TLTC   -> True
+  ZEC    -> False
+  TZEC   -> True
+  CPR    -> False
+  TCPR   -> True
+  DASH   -> False
+  TDASH  -> True
```

Comment:
These functions probably should replace `currencyCodeToCurrency` and `currencyToCurrencyCode` from Ergvein.Index.Server.TCPService.Conversions module